crates/ty/docs/configuration.md

```diff
@@ -153,6 +153,64 @@ typeshed = "/path/to/custom/typeshed"
 
 ## `src`
 
+#### `exclude`
+
+A list of file and directory patterns to exclude from type checking.
+
+Patterns follow a syntax similar to `.gitignore`:
+- `./src/` matches only a directory
+- `./src` matches both files and directories
+- `src` matches files or directories named `src` anywhere in the tree (e.g. `./src` or `./tests/src`)
+- `*` matches any number of characters except `/`
+- `**` matches any number of characters including `/`
```

Comment:
I think I would be tempted to word this as, "`**` matches zero or more path components." So like, `**a` is kind of a weird glob. I think `globset` does currently accept that and does implement the semantics you describe here though.